src/dev/RedundantADC.cpp

```diff
@@ -0,0 +1,69 @@
+#include <EVT/io/ADC.hpp>
+#include <cmath>
+#include <dev/RedundantADC.hpp>
+
+namespace IO = EVT::core::IO;
+
+constexpr uint32_t LOW_MARGIN = 1;
+constexpr uint32_t HIGH_MARGIN = 5;
+
+namespace HIB::DEV {
+
+RedundantADC::RedundantADC(IO::ADC& adc0, IO::ADC& adc1, IO::ADC& adc2) : adc0(adc0), adc1(adc1), adc2(adc2) {}
+
+RedundantADC::Status RedundantADC::readVoltage(uint32_t& return_val) {
+    // Read ADC values
+    int32_t adcValues[3];
+    adcValues[0] = static_cast<int32_t>(adc0.read() * 1000);
+    adcValues[1] = static_cast<int32_t>(adc1.read() * 1000);
+    adcValues[2] = static_cast<int32_t>(adc2.read() * 1000);
+
+    // Calculate average of all ADC values
+    int32_t average = (adcValues[0] + adcValues[1] + adcValues[2]) / 3;
+
+    // Check for margin error
+    bool adc0underLow = (std::abs(adcValues[0] - average) * 100 / average) < LOW_MARGIN;
+    bool adc1underLow = (std::abs(adcValues[1] - average) * 100 / average) < LOW_MARGIN;
+    bool adc2underLow = (std::abs(adcValues[2] - average) * 100 / average) < LOW_MARGIN;
+
+    bool adc0underHigh = (std::abs(adcValues[0] - average) * 100 / average) > HIGH_MARGIN;
```

Comment:
I spent a while looking at this again, and I'm pretty sure these are actually supposed to be `<`.
```suggestion
    bool adc0underHigh = (std::abs(adcValues[0] - average) * 100 / average) < HIGH_MARGIN;
```